core-build/test-web-library-build/jest.config.json

```diff
@@ -0,0 +1,8 @@
+{
+  "testPathIgnorePatterns": [
+    "<rootDir>/(src|lib-amd|lib-es6|coverage|build|docs|node_modules)/"
+  ],
+  "testMatch": [
+    "**/*.test.js?(x)"
+  ]
+}
```

Comment:
Can this go in the /config folder?

Reply:
When we enable Jest for real projects, ideally there should be no config changes required.  Or if you want opt-in, it should be something like "useJest=true" in one of the existing files.

Using Jest definitely should NOT involve:

- Modifying the Gulpfile

- Adding nonstandard files outside of the ./config folder

- Copying+pasting nontrivial boilerplate into every single project that uses Jest (e.g. what if we need to update these test paths in the future?)

*Ideally* we should also be able to do this without each project having to add Jest to its package.json and tsconfig.json

The whole spirit of GCB is to avoid copying+pasting tons of stuff into every single project, and then having to maintain all those copies whenever a fix is needed.

---
In reply to: [145567960](https://github.com/Microsoft/web-build-tools/pull/391#discussion_r145567960) [](ancestors = 145567960)

Reply:
The project that wants to use Jest needs to include types/Jest in the package.json (just like types/mocha etc) and remove types/mocha. We cannot avoid that.

---
In reply to: [145592254](https://github.com/Microsoft/web-build-tools/pull/391#discussion_r145592254) [](ancestors = 145592254,145567960)